[Refactor][Documentation] Update Biome README and do more refactoring (#976)
Despair-Games/poketernity: src/data/biomes/README.md

#### Pokemon Pool
A biome's `pokemonPool` consists of a mapping of a `BiomePoolTier` (common, uncommon, rare, super rare, ultra rare, boss, boss rare, boss super rare, boss ultra rare) to time of day (dawn, day, dusk, night, all), and finally to the pokemon that appear. The current time of day is combined with the ALL pool to determine what Pokemon can appear. The rarity is rolled first, and if there are no available selections in that tier, it is automatically downgraded according to the above order in BiomePoolTier.

> For the most part, there is no need to specify evolutions in the Pokemon pool since Pokemon will automatically evolve or de-evolve based on level. The only reason to specify an evolved form is wanting to guarantee that form shows up. (Specifying Gardevoir means Ralts, Kirlia, and Gardevoir can appear, but not Gallade). For Pokemon with specific forms, check the bottom section for how to determine that based on biome.

Reviewer: NIT: `Check the bottom section` could be an internal link to that section

Author: The reason I didn't link it was because I didn't remember how to 😂

#### Pokemon Pool Philosophy

Sam's general design philosophy had all the boss tiers contain Pokemon one tier higher than the non boss tier. `BOSS` would use some Pokemon from `COMMON`, `UNCOMMON`, and `RARE`; `BOSS_RARE` would use some Pokemon from `RARE` and `SUPER_RARE`; `BOSS_SUPER_RARE` would use Pokemon from `ULTRA_RARE`; and `BOSS_ULTRA_RARE` would be reserved for the truly rare Pokemon (usually box legends).

> Here is a basic definition for what Pokemon belong in what pools

Reviewer: Q: Is `what` correct here? Shouldn't it be `which`?

The loop point of a bgm, measured in seconds, should be set to ensure that the bgm loops/replays smoothly after the bgm file's ending.

#### locale

> The biome needs to have a proper locale mapping defined in [https://github.com/Despair-Games/poketernity-locales/tree/main](https://github.com/Despair-Games/poketernity-locales/tree/main) or the correct `biome.json` file or else the biome name will be displayed as the biomeId enum in all caps

Reviewer: ```suggestion
The biome needs to have a proper locale mapping defined in either the [main](https://github.com/Despair-Games/poketernity-locales/tree/main) branch (locales repo) or the correct `biome.json` file. If not the biome name will be displayed as the biomeId enum in all caps (e.g. `"TOWN"`)
```

Not sure I understand this correct. What's the `biome.json` here? is that a file in the locales repo?

Author: Yes, `biome.json` is a file that lives in the locales repo

### Other code that needs updating
* Don't forget to add connections to and from the biome in `biome-link.ts`
* For now, don't forget to update `biome-utils.indoorBiomes` and `arena.biomeWithProps` to account for tinting and prop display
* For Pokemon with specific forms in certain biomes, `arena.getSpeciesFormIndex` can set specific forms for specific biomes
* Alcremie requires a specific biome to determine its form and those biomes are defined in `biome-utils.ts`
* There are also a few arrays of biomes used to determine specific MEs that need to be updated in `biome-utils.ts`

> * The move Camoflage uses the [CopyBiomeTypeAttr](../moves/move-attrs/copy-biome-type-attr.ts) which changes the user's type based on the current biome

Reviewer 2: ```suggestion
* The move Camouflage uses the [CopyBiomeTypeAttr](../moves/move-attrs/copy-biome-type-attr.ts) which changes the user's type based on the current biome
```

* The move Secret Power uses the [SecretPowerAttr](../moves/move-attrs/secret-power-attr.ts) which applies a secondary effect based on the current biome

> * The move Nature Power uses the [NaturePowerAttr](../moves/move-attrs/nature-power-attr.ts) which changes into a different attack depending on the current biome

Author: I was also considering moving these 3 mappings into `biome-utils.ts` but didn't know if that would bloat the file too much

### Other code that needs updating
* Don't forget to add connections to and from the biome in `biome-link.ts`
* In order to enable MEs, the biome needs to be added to `mystery-encounter.mysteryEncountersByBiome` along with a list of MEs that are enabled. If the new biome falls under the categories of `EXTREME`, `NON_EXTREME`, `HUMAN_TRANSITABLE`, or `CIVILIZATION`, be sure to add the Biome to the corresponding lists in `biome-utils.ts` to enable MEs with those specific categories

> * For now, don't forget to update `biome-utils.indoorBiomes` and `arena.biomeWithProps` to account for tinting and prop display

Reviewer: ```suggestion
* Don't forget to update `biome-utils.indoorBiomes` and `arena.biomeWithProps` to account for tinting and prop display
```

"For now" isn't a relevant information *for now*

* For Pokemon with specific forms in certain biomes, `arena.getSpeciesFormIndex` can set specific forms for specific biomes
* Alcremie requires a specific biome to determine its form and those biomes are defined in `biome-utils.ts`

> * There are also a few arrays of biomes used to determine specific MEs that need to be updated in `biome-utils.ts`

Reviewer: NIT: Link to `biome-utils.ts` file in the `beta` branch (don't use permalink)
